src/stirling/source_connectors/socket_tracer/protocols/mux/types.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <magic_enum.hpp>
+#include "src/stirling/source_connectors/socket_tracer/protocols/common/event_parser.h"
+#include "src/stirling/utils/utils.h"
+
+namespace px {
+namespace stirling {
+namespace protocols {
+namespace mux {
+
+enum class Type : int8_t {
+    Treq = 1,
+    Rreq = -1,
+    Tdispatch = 2,
+    Rdispatch = -2,
+
+    // control messages
+    Tdrain = 64,
+    Rdrain = -64,
+    Tping  = 65,
+    Rping  = -65,
+
+    Tdiscarded = 66,
+    Rdiscarded = -66,
+
+    Tlease = 67,
+
+    Tinit = 68,
+    Rinit = -68,
+
+    Rerr = -128,
+
+    // only used to preserve backwards compatibility
+    TdiscardedOld = -62,
+    RerrOld       = 127,
+};
+
+inline bool IsMuxType(int8_t t) {
+  std::optional<Type> mux_type = magic_enum::enum_cast<Type>(t);
+  return mux_type.has_value();
+}
+
+inline Type GetMatchingRespType(Type req_type) {
+    if (req_type == Type::RerrOld) return Type::RerrOld;
+    if (req_type == Type::Rerr) return Type::Rerr;
+    if (req_type == Type::Tinit) return Type::Rinit;
+    if (req_type == Type::Tping) return Type::Rping;
+    if (req_type == Type::Treq) return Type::Rreq;
+    if (req_type == Type::Tdrain) return Type::Rdrain;
+    if (req_type == Type::Tdispatch) return Type::Rdispatch;
+    if (req_type == Type::TdiscardedOld || req_type == Type::Tdiscarded) return Type::Rdiscarded;
+    return Type::Tlease;
```

Comment:
This shouldn't return `Type::Tlease` by default otherwise new mux message types will be incorrectly consumed. Is there a pattern in this code base to return an optional type? It seems like `StatusOr<Type>` might fit?

If we don't find a match we want to ignore the message.